### PR TITLE
Parse instanced static meshes' data to retrieve instances

### PR DESCRIPTION
**Pull request by alex4401**
Creating an MR to have reviews in one place, as this will need them, touching the `ue` module deeply.

- [x] Parse data in ISMCs located after the property table (i.e. in addition to standard UObject data)
- [x] Export the harvestables locations in wiki.maps
- - [x] Determine an output path
- [x] Add missing pieces that break parsing of client assets (likely lightmap/shadowmap references?)

*Project note description:*
* Every resource node is an instance of a static mesh. These instances are handled by Hierarchical Instanced Static Meshes, which is expands on the basic UObject's serialization code.
* To read the required data from HISMCs, we have to read data that is after the property table.
* We only need to read the ISM part.
* ISM has instance transforms we're looking for. They're represented by a 4x4 matrix.
* First resource map made with data from this branch is going live soon on wiki (Genesis Part 1 and Genesis Part 2 maps).

### CODE REVIEW

**Comment by alex4401** [comment]
Have fun with this.

**Review by coldino — COMMENTED** [review]
Core stuff looking good, and I know you're already changing the way the resources are output and configured. Should be an easy merge after that.

File: ue/objects.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    self.stream.offset += 3*4 + 4*4 + 1
+
+        # Instances are serialised as a memory dump - UE precedes this with size of the struct for safety (to prevent loading an
+        # incompatible asset if the struct layout has changed).
+        assert self.stream.readUInt32() == 80
+        num_instances = self.stream.readUInt32()
+        instances = []
+
+        for index in range(num_instances):
+            # TODO: check if instance is visible (i.e. 1 in the bitmask)
```
[the remark this answers is not in the document]
Comment (alex4401) [reply ue/objects.py:52]: *Fuck.*

File: export/wiki/maps/world.py [hunk 2]
```diff
@@ -116,22 +136,31 @@ def construct_export_files(self):
...
+    '''
+    Finds a data gathering helper for an export.
+    '''
+    # Retrieve parents if possible. Skip export otherwise.
     try:
         parents = set(find_parent_classes(export, include_self=True))
     except (AssetLoadException, MissingParent):
         return None
 
+    # Find a helper that can accept the export.
```
[the remark this answers is not in the document]
Comment (alex4401) [reply export/wiki/maps/world.py:163]: .

**Comment by alex4401** [comment]
This will not be continued.

**Pull request closed without merging**